### PR TITLE
Development fx area and volume

### LINKED ISSUE #367: Preprocessor area_statistics broken when using fx_files
Example recipe:
```yaml
datasets:
  - {dataset: HadGEM2-ES, project: CMIP5, exp: historical, ensemble: r1i1p1, start_year: 2001, end_year: 2004}

preprocessors:
  global_total:
    area_statistics:
      operator: sum
      fx_files: [areacello]

diagnostics:
  test:
    scripts: null
    variables:
      fgco2:
        preprocessor: global_total
        mip: Omon
 ```

Stack trace:
```
Traceback (most recent call last):
  File "/home/bandela/conda/envs/esmvaltool/lib/python3.7/multiprocessing/pool.py", line 121, in worker
    result = (True, func(*args, **kwds))
  File "/home/bandela/src/esmvalcore/esmvalcore/_task.py", line 686, in _run_task
    output_files = task.run()
  File "/home/bandela/src/esmvalcore/esmvalcore/_task.py", line 238, in run
    self.output_files = self._run(input_files)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 403, in _run
    product.apply(step, self.debug)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 271, in apply
    self.cubes = preprocess(self.cubes, step, **self.settings[step])
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 216, in preprocess
    result.append(_run_preproc_function(function, item, settings))
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/__init__.py", line 199, in _run_preproc_function
    return function(items, **kwargs)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/_area.py", line 212, in area_statistics
    grid_areas = tile_grid_areas(cube, fx_files)
  File "/home/bandela/src/esmvalcore/esmvalcore/preprocessor/_area.py", line 152, in tile_grid_areas
    grid_areas = da.tile(grid_areas, [cube.shape[0], 1, 1])
  File "/home/bandela/conda/envs/esmvaltool/lib/python3.7/site-packages/dask/array/creation.py", line 800, in tile
    raise NotImplementedError("Only integer valued `reps` supported.")
NotImplementedError: Only integer valued `reps` supported.
"""
```

### PR DESCRIPTION
**Pull request by valeriupredoi**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValTool/blob/version2_development/CONTRIBUTING.md).

**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [ ] This pull request has a descriptive title that can be used in a changelog
-   [ ] Add unit tests
-   [ ] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [ ] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes #367

### CODE REVIEW

**Comment by ledm** [comment]
Thanks V!

**Comment by bouweandela** [comment]
Related issue #94, see also the discussion there.

**Comment by valeriupredoi** [comment]
@ledm would you be ok if you merged `development` (fix the conflicts) and then crack on with the review for this pls - it's something that shouldn't be lost in the cracks :beer:

File: esmvalcore/preprocessor/_area.py [hunk 1]
```diff
@@ -56,20 +57,24 @@ def extract_region(cube, start_longitude, end_longitude, start_latitude,
...
+    # Regular grid
     if cube.coord('latitude').ndim == 1:
         region_subset = cube.intersection(
             longitude=(start_longitude, end_longitude),
             latitude=(start_latitude, end_latitude))
         region_subset = region_subset.intersection(longitude=(0., 360.))
         return region_subset
-    # irregular grids
+
+    # Irregular grids - not lazy.
```
Comment (bouweandela) [inline esmvalcore/preprocessor/_area.py]: These changes are no longer needed, this has been fixed.

**Comment by valeriupredoi** [comment]
this needs revival :beer:

**Comment by valeriupredoi** [comment]
Nope! these changes are obsolete, all is done up in `development` already! Yay :tada:

**Comment by bouweandela** [comment]
This problem is not solved: https://github.com/ESMValGroup/ESMValCore/issues/367

**Comment by valeriupredoi** [comment]
s**t, sorry mate

The branch is old as time and I am afraid I don't have the time to resolve all the conflicts in `_area.py` (and the expertise either since this has @ledm written all over it) -> Lee the floor is yours here man :beer:

**Comment by valeriupredoi** [comment]
dis overarched by #439 and #446

**Comment by mattiarighi** [comment]
Please close and delete branch if no longer required.

**Pull request closed without merging**